src/actions.sol

```diff
@@ -50,30 +64,34 @@ contract Actions {
         borrowWithdraw(shelf, loan, amount, usr);
     }
 
-    function issueLockBorrowWithdraw(ShelfLike shelf, address registry, uint token, uint amount, address usr) public {
-        uint loan = shelf.issue(registry, token);
+    function transferIssueLockBorrowWithdraw(ShelfLike shelf, NFTLike registry, uint token, uint amount, address usr) public {
+        uint loan = transferIssue(shelf, registry, token);
         lockBorrowWithdraw(shelf, loan, amount, usr);
     }
 
-    function repay(ShelfLike shelf, uint loan, uint amount) public {
+    function repay(ShelfLike shelf, ERC20Like erc20, uint loan, uint amount) public {
+        erc20.approve(address(shelf), amount);
+        // transfer money from borrower to proxy
+        erc20.transferFrom(msg.sender, address(this), amount);
         shelf.repay(loan, amount);
     }
 
-    function unlock(ShelfLike shelf, uint loan) public {
+    function unlock(ShelfLike shelf, NFTLike registry, uint token, uint loan) public {
         shelf.unlock(loan);
+        registry.transferFrom(address(this), msg.sender, token);
```

Comment:
here theoretically we could allow more flexibility by providing an `usr` instead of `msg.sender`

Reply:
we can add that later.